sympy/tensor/array/arrayop.py

```diff
@@ -279,6 +281,16 @@ def permutedims(expr, perm):
 
     # Get the inverse permutation:
     iperm = ~perm
+    new_shape = perm(expr.shape)
+
+    if isinstance(expr, SparseNDimArray):
+        new_array = {}
+        for k, v in expr._sparse_array.items():
+            index = 0
+            for i, idx in enumerate(perm(expr._get_tuple_index(k))):
+               index = index * new_shape[i] + idx
+            new_array[index] = v
```

Comment:
are you sure you can't do this in one line? I believe you can use a dictionary comprehension.

Reply:
I am afraid not. the problem is in the inner loop. The inner loop sums up terms in a succesive way, so I couldn't see how dict comprehension can be used in this case. Even `Sum()` is not applicable here I think.
In fact the inner loop have the same functionality as `_parse_index` of NDimArray. One alternative solution would be:
```python
# Initializing a mutable sparse array at first
temp_arr = MutableSparseNDimArray({}, new_shape)
temp_arr._sparse_array = {temp_arr._parse_index(perm(expr._get_tuple_index(k))):v 
                              for k,v in expr._sparse_array.items()}
if isinstance(expr, Immutable):
    return temp_arr.as_immutable()
return temp_arr
```
But the code is not very readable and it will create an unnecessary new `mutable` object if expr is `immutable`
What do you think of it?

Reply:
what about:

```python
return type(expr)(
    {temp_arr._parse_index(perm(expr._get_tuple_index(k))): v for k, v in expr._sparse_array.items()},
    new_shape
)
```